debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -151,6 +187,42 @@ protected void initPublication() {
         }
     }
 
+    private Stream<TableId> toTableIds(Set<TableId> tableIds, Pattern pattern) {
+        return tableIds
+                .stream()
+                .filter(tid -> pattern.asPredicate().test(tid.toString()))
+                .sorted();
+    }
+
+    private Set<TableId> determineCapturedTables() throws Exception {
+        Set<TableId> allTableIds = this.connect().readTableNames(pgConnection().getCatalog(), null, null, new String[]{ "TABLE" });
+
+        Set<TableId> capturedTables = new HashSet<>();
+
+        String value = config().getString(RelationalDatabaseConnectorConfig.TABLE_WHITELIST);
+
+        for (TableId tableId : allTableIds) {
+            if (tableFilter.dataCollectionFilter().isIncluded(tableId)) {
+                LOGGER.trace("Adding table {} to the list of captured tables", tableId);
+                capturedTables.add(tableId);
+            }
+            else {
+                LOGGER.trace("Ignoring table {} as it's not included in the filter configuration", tableId);
+            }
+        }
+
+        if (value != null) {
```

Comment:
Hum, I think I've asked about it before, but why is this explicit check against TABLE_WHITELIST done here? It shouldn't be needed, given you already apply the data collection filter above. Is it some sort of left-over perhaps?

Reply:
yeah this does seem like a left over!